### PR TITLE
Fix VSCode dev container integration

### PR DESCRIPTION
**Pull request by chipn**
### Fixes: #51 

This PR updates the VSCode development container integration to work properly on both Windows 10/WSL2 and Mac including Run/Debug.

The following changes were implemented:

1. Fix the debug Dockerfile command to properly sleep and allow Nautobot to be started/stopped without the container being stopped.
2. Adjust paths for use within the container.
3. Add feature to prompt opening a browser when Nautobot is started via Run/Debug.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -8,14 +8,12 @@
 		"eamodio.gitlens",
 		"oderwat.indent-rainbow",
 		"ms-python.python",
-		"ms-vsliveshare.vsliveshare",
-		"littlefoxteam.vscode-python-test-adapter"
```
Comment (jvanderaa) [inline .devcontainer/devcontainer.json:12]: Why do we want to remove the test explorer? I'd love to see this kept in.

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -8,14 +8,12 @@
...
+		"ms-vsliveshare.vsliveshare"
 	],
 	"settings": {
 		"python.pythonPath": "/usr/local/bin/python",
 		"python.analysis.extraPaths": ["/opt/nautobot"],
 		"python.linting.pylintEnabled": true,
-		"python.linting.enabled": true,
-		"pythonTestExplorer.testFramework": "pytest"
```
Comment (jvanderaa) [inline .devcontainer/devcontainer.json:19]: Keep the testing framework if possible?

**Review by jvanderaa — COMMENTED** [review]
I'd like to explore keeping of the testing modules. They provide a good methodology to be able to see visually what tests may be failing.

**Comment by smk4664** [comment]
So, I think one of my commit messages got lost in Chip's rebase.  We are not using nor installing pytest in this repo.  The pythonTestExplorer is a plugin that only supports pytest or unittest.  If you are using the devcontainers and try to use the plugin, it fails due to pytest missing.

**Comment by jvanderaa** [comment]
This makes sense. There was not a mention of the reason for removing of the testing tools installation. My comments with that update are no longer relevant.

**Comment by chipn** [comment]
> This makes sense. There was not a mention of the reason for removing of the testing tools installation. My comments with that update are no longer relevant.

Stephen's commit message is here:

https://github.com/nautobot/nautobot/pull/73/commits/1243d2463c570ff425ed099b0884bf931a932f8e

**Comment by smk4664** [comment]
> I'd like to explore keeping of the testing modules. They provide a good methodology to be able to see visually what tests may be failing.

I agree with this, but we currently do not use pytest, so this plugin is useless, and takes up loading time.

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (smk4664) [reply .devcontainer/devcontainer.json:19]: This configuration is for the Plugin, so if we remove the plugin, we should remove the configuration.

**Comment by chipn** [comment]
> I tried rebasing under this changeset, and it seems to me like the Nautobot server doesn't actually get started if I launch the container from within VSCode, possibly because of the "sleep infinity" in docker-compose.debug.yml. If I start the container manually (`invoke start`) _then_ connect to it from VSCode, then the server runs properly and I can also run commands from within VSCode. Am I doing something wrong in the first case, or should we change/remove `docker-compose.debug.yml`?

That's expected as we don't want the container to stop if Nautobot is stopped. Once you connect to the container in VSCode, you can start/stop Nautobot either via the Run menu or directly in the terminal. I tried to document that in the doc changes for this PR, but perhaps it wasn't clear.

**Comment by glennmatthews** [comment]
> I tried rebasing under this changeset, and it seems to me like the Nautobot server doesn't actually get started if I > That's expected as we don't want the container to stop if Nautobot is stopped. Once you connect to the container in VSCode, you can start/stop Nautobot either via the Run menu or directly in the terminal. I tried to document that in the doc changes for this PR, but perhaps it wasn't clear.

You're absolutely right, it is documented. :-) My apologies. In that case this seems to work quite well and is a clear improvement over the existing integration. Thanks!

**Review by glennmatthews — APPROVED** [review]

**Review by jathanism — APPROVED** [review]

**Pull request merged**